pkg/identityManager/client.go

```diff
@@ -15,59 +15,66 @@
 package identitymanager
 
 import (
-	"net/http"
-	"net/url"
+	"context"
+	"fmt"
 
 	corev1 "k8s.io/api/core/v1"
-	kerrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/rest"
-	"k8s.io/klog/v2"
+	"k8s.io/client-go/tools/clientcmd"
 
 	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	"github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/utils/getters"
 )
 
 // GetConfig gets a rest config from the secret, given the remote clusterID and (optionally) the namespace.
 // This rest config con be used to create a client to the remote cluster.
-func (certManager *identityManager) GetConfig(remoteCluster discoveryv1alpha1.ClusterIdentity,
-	namespace string) (*rest.Config, error) {
-	var secret *corev1.Secret
-	var err error
+func (certManager *identityManager) GetConfig(remoteCluster discoveryv1alpha1.ClusterIdentity, _ string) (*rest.Config, error) {
+	ctx := context.TODO()
 
-	if namespace == "" {
-		secret, err = certManager.getSecret(remoteCluster)
-	} else {
-		secret, err = certManager.getSecretInNamespace(remoteCluster, namespace)
-	}
+	// Get Identity of type ControlPlane associated to the given remote cluster.
+	identity, err := getters.GetControlPlaneIdentityByClusterID(ctx, certManager.client, remoteCluster.ClusterID)
```

Comment:
Can we directly get the secret in order to avoid dependencies with the authentication module?